### DISCUSSION TITLE
How to use private Git repos as npm dependencies?

### DISCUSSION THREAD

**Discussion by richardsimko** [renovatebot/renovate#22344] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@39.0.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate should be able to authenticate in order to install an npm dep such as `"private-dep": "git+ssh://git@github.com:org/private-dep.git"`.

If we're able to rewrite the SSH URL to HTTPS that would be fine as we can pass the auth token. Passing an SSH key for renovate to use would also work.

### What you tried so far.

* Setting the auth in GH Actions using `git config --global url."https://${{ secrets.NPM_TOKEN }}@github.com/".insteadOf "git@github.com:"` (This works for all other GH actions)
* Setting the auth in renovate using:
```
 "customEnvVariables": {
    "GIT_CONFIG_COUNT": "1",
    "GIT_CONFIG_KEY_0": "url.https://${NPM_TOKEN}@github.com/.insteadOf",
    "GIT_CONFIG_VALUE_0": "git@github.com:"
  },
```
* Disabling the host entirely using
```
"hostRules": [
    {
      "matchHost": "git@github.com",
      "enabled": false
    }
  ],
```

### Relevant debug logs

<details><summary>Logs</summary>

```
"stderr": "error Command failed.\nExit code: 128\nCommand: git\nArguments: ls-remote --tags --heads git@github.com:org/private-repo.git\nDirectory: /tmp/renovate/repos/github/org/project\nOutput:\nHost key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n"
```

</details>

**Comment by rushilsrivastava** [comment]
Were you able to find a solution to this?
